### PR TITLE
Use config file in `reek` matcher

### LINKED ISSUE #1093: reek/spec rspec helper not picking up config.reek
I tested with the following in `config.reek` file in my home directory.

``` yaml

---
"app/controllers":
  IrresponsibleModule:
    enabled: false
  NestedIterators:
    max_allowed_nesting: 2
  UnusedPrivateMethod:
    enabled: false
  InstanceVariableAssumption:
    enabled: false
"app/helpers":
  IrresponsibleModule:
    enabled: false
  UtilityFunction:
    enabled: false
exclude_paths:
  - db
  - config
```

Then in my rails_helper.rb

``` ruby
require 'reek/spec'
```

Then in my `spec/smells/reek_spec.rb` file

``` ruby
require 'rails_helper'

describe 'Code Smells' do
  # Check all ruby files in lib and app directory
  file_patterns = { lib: 'lib/**/*.rb', app: 'app/**/*.rb' }
  file_patterns.each do |directory, file_pattern|

    # within the context of given directory
    context "with #{directory}/ files" do
      Pathname.glob(file_pattern).sort.each do |file|

        # Execute smell check for each file
        context "with #{file} file" do
          it 'contains no code smells' do
            expect(file).not_to reek
          end
        end
      end
    end
  end
end
```

When I run my specs, I get a failure:

``` Shell
$ rspec spec/smells/reek_spec.rb
Failures: 
  1) Sniff Tests with app/ files with app/controllers/foo_bars_controller.rb file contains no code smells
     Failure/Error: expect(file).not_to reek

       Expected no smells, but got:
         InstanceVariableAssumption: FooBarsController assumes too much for instance variable '@foobar'
```

However when I run reek at the command line, I get no errors.

``` Shell
$ reek
0 total warnings
```

### PR DESCRIPTION
**Pull request by mvz**
Fixes #1093.

### CODE REVIEW

File: features/rspec_matcher.feature [hunk 1]
```diff
@@ -0,0 +1,43 @@
+Feature: Use reek_of mathcer
```
Comment (troessner) [inline features/rspec_matcher.feature]: Typo

File: features/rspec_matcher.feature [hunk 2]
```diff
@@ -0,0 +1,43 @@
...
+
+  Scenario: Pick up configuration file
+    Given the smelly file 'smelly.rb'
+    And a configuration file 'full_mask.reek'
+    And a file "reek_spec.rb" with:
+    """
+    require 'reek/spec'
+
+    describe 'smelly.rb' do
+      it 'is fragrant' do
```
Comment (troessner) [inline features/rspec_matcher.feature]: Should "is NOT" i guess?

File: features/rspec_matcher.feature [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+        expect(Pathname.new('smelly.rb')).not_to reek
+      end
+    end
+    """
+    When I run `rspec reek_spec.rb`
+    Then it succeeds
+    And stdout should contain:
```
Comment (troessner) [inline features/rspec_matcher.feature]: The second scenario is almost identical with a couple of easy to miss differences. Would be great if we could make the differences more obvious.

File: features/rspec_matcher.feature (hunk 1 above)
Comment (mvz) [reply features/rspec_matcher.feature]: Fixed

File: features/rspec_matcher.feature (hunk 3 above)
Comment (mvz) [reply features/rspec_matcher.feature]: I've changed the scenarios around so I could extract more common parts. Let me know what you think.

**Comment by troessner** [comment]
Yes, I like those cucumber scenarios a lot better, merged!

**Pull request merged**